src/modules/function/declaration.rs

```diff
@@ -271,13 +277,74 @@ impl TranslateModule for FunctionDeclaration {
 impl DocumentationModule for FunctionDeclaration {
     fn document(&self, meta: &ParserMetadata) -> String {
         let mut result = vec![];
-        result.push(format!("## `{}`", self.name));
+        result.push(format!("## `{}`\n", self.name));
+
+        let references = self.create_reference(meta, &mut result);
```

Comment:
I moved the new code to a helper function, which returns an optional string vector (encapsulating both the success of the function, and the additional lines to be added at end of section).